src/sandstorm/web-session.capnp

```diff
@@ -154,12 +232,16 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
       created  @1 $httpStatus(id = 201, title = "Created");
       accepted @2 $httpStatus(id = 202, title = "Accepted");
 
+      noContent      @3 $httpStatus(id = 204, title = "No Content");
+      partialContent @4 $httpStatus(id = 206, title = "Partial Content");
+      multiStatus    @5 $httpStatus(id = 207, title = "Multi-Status");
+
+      # This seems to fit better here than in the 3xx range
+      notModified    @6 $httpStatus(id = 304, title = "Not Modified");
```

Comment:
How about we have a completely separate entry in the union -- as a sibling of `content` -- like:

    preconditionFailed @22 :Void;

We use this both for the traditional "precondition failed" and "not modified" responses -- we can determine which one was meant in the front-end by checking the request method and precondition: for the specific combination of GET + If-None-Match, we return a 304, otherwise we return a 412.

I think this make a lot of sense because `preconditionFailed` is semantically different from other error cases, and `notModified` is semantically different from success cases. HTTP's taxonomy of status codes is kind of ad hoc and we don't necessarily need to follow it here. And the only reason why HTTP uses status code 304 rather than always 412 is for backwards-compatibility with the time-based caching in HTTP/1.0.